File: transactions/factory.go
package session

import (
	"github.com/alvidir/mastermind/service/session/transactions"
	"github.com/alvidir/util/pattern/transaction"
)

func newTxLogin() transaction.Tx {
	body := &transactions.TxLogin{}
	return transaction.NewTransaction(body)
}

func newTxGoogleLogin() transaction.Tx {
	body := &transactions.TxGoogleLogin{}
	return transaction.NewTransaction(body)
}

func newTxLogout() transaction.Tx {
	body := &transactions.TxLogout{}
	return transaction.NewTransaction(body)
}

func newTxSignup() transaction.Tx {
	body := &transactions.TxSignup{}
	return transaction.NewTransaction(body)
}

File: src/service/session/transactions/factory.rs
use crate::transaction::traits::Tx;
use crate::transaction::factory as TxFactory;
use crate::service::session::transactions::{login::TxLogin,
                                            signup::TxSignup};

pub fn new_tx_signup(name: String, addr: String, pwd: String) -> Box<dyn Tx> {
    let body = TxSignup::new(name, addr, pwd);
    TxFactory::new_transaction(Box::new(body))
}

pub fn new_tx_login(cookie: String, name: String, addr: String, pwd: String) -> Box<dyn Tx> {
    let body = TxLogin::new(cookie, name, addr, pwd);
    TxFactory::new_transaction(Box::new(body))
}